.github/workflows/build-package-docs.yaml

```diff
@@ -120,3 +132,77 @@ jobs:
         name: package-docs-build
         path: build-directory/docs/docsite/ansible-package-docs-html-*.tar.gz
         retention-days: 7
+
+  deploy-package-docs:
+    if: fromJSON(github.event.inputs.deploy)
+    needs: build-package-docs
+    runs-on: ubuntu-latest
+    environment:
+      name: deploy-package-docs
+      url: https://github.com/ansible/ansible-documentation
+    env:
+      dest-repo: owner/repo-name  # TODO create repo and specify here
+      dest-branch: main  # TODO specify repo branch here
+      user-email: "41898282+github-actions[bot]@users.noreply.github.com"
+      user-name: "github-actions[bot]"
+    steps:
+    - name: Download the build archive
+      uses: actions/download-artifact@v4
+      with:
+        name: package-docs-build
+
+    - name: Extract the tarball
+      run: |
+        set -xeEuo pipefail
+        mkdir -pv extracted-docs
+        tar -xvzf ansible-package-docs-html-*.tar.gz -C extracted-docs
+
+    - name: Generate temp GITHUB_TOKEN
+      id: create_token
+      uses: tibdex/github-app-token@v2
+      with:
+        app_id: ${{ secrets.BOT_APP_ID }}  # TODO create bot app id secret
+        private_key: ${{ secrets.BOT_APP_KEY }}  # TODO create bot app id key
```

Comment:
It might be easier to use a Github deploy key rather than mucking around with app tokens if all you need is push access.

Reply:
Thanks for the comment @gotmax23 ! Yes, I think we're going to need to use a deploy key instead of an app. I installed a new app in `ansible` but if we're going to transfer across orgs to a repo in `ansible-community` then it gets complicated.

I kind of wanted to get some other thoughts/opinions on this, so cheers.

Reply:
Yeah, let's just use a deploy key. I've updated the PR and tested it on my forks. Thanks again.